File: QViz.NET/Periscope.QViz/JSON/APIQuery.cs
using System;

namespace Periscope.QViz.JSON
{
	/// <summary>
	/// JSON Model for Test API Query Parameter Details
	/// </summary>
	public class APIQuery
	{
		/// <summary>
		/// Identifier of the API Query Object
		/// </summary>
		public string apiQueryId;
		
		/// <summary>
		/// Serial Number of the API Query Object
		/// </summary>
		public int srNo;
		
		/// <summary>
		/// Identifier of the API Object
		/// </summary>
		public string apiId;
		
		/// <summary>
		/// Identifier of the API Query Value Object
		/// </summary>
		public string queryValueId;
		
		/// <summary>
		/// Date and Time on which this Model was Created
		/// </summary>
		public DateTime createdOn;
		
		/// <summary>
		/// Date and Time on which this Model was last Updated
		/// </summary>
		public DateTime updatedOn;
		
		/// <summary>
		/// Value of the API Query Object
		/// </summary>
		public APIQueryValue queryValue;

		/// <summary>
		/// Default Constructor
		/// </summary>
		public APIQuery()
		{
			this.apiQueryId = "";
			this.srNo = 0;
			this.apiId = "";
			this.queryValueId = "";
			this.createdOn = DateTime.Now;
			this.updatedOn = DateTime.Now;
			this.queryValue = new APIQueryValue();
		}
	}
}

File: QViz.Java/src/main/java/com/periscope/qviz/json/APIQuery.java
package com.periscope.qviz.json;

import java.util.Date;

/**
 * JSON Model for Test API Query Parameter Details
 */
public class APIQuery {

	/**
	 * Identifier of the API Query Object
	 */
	public String apiQueryId;

	/**
	 * Serial Number of the API Query Object
	 */
	public Integer srNo;

	/**
	 * Identifier of the API Object
	 */
	public String apiId;

	/**
	 * Identifier of the API Query Value Object
	 */
	public String queryValueId;

	/**
	 * Date and Time on which this Model was Created
	 */
	public Date createdOn;

	/**
	 * Date and Time on which this Model was last Updated
	 */
	public Date updatedOn;

	/**
	 * Value of the API Query Object
	 */
	public APIQueryValue queryValue;

	/**
	 * Default Constructor
	 */
	public APIQuery() {
		this.apiQueryId = "";
		this.srNo = 0;
		this.apiId = "";
		this.queryValueId = "";
		this.createdOn = null;
		this.updatedOn = null;
		this.queryValue = new APIQueryValue();
	}

}
